plugins/de.cau.cs.kieler.klighd.lsp/src/de/cau/cs/kieler/klighd/lsp/structuredProgramming/sccharts/StructuredProgScchartLanguageServerExtension.xtend

```diff
@@ -0,0 +1,729 @@
+ * 
+ * http://rtsys.informatik.uni-kiel.de/kieler
+ * 
+ * Copyright ${year} by
+ * + Kiel University
+ *   + Department of Computer Science
+ *     + Real-Time and Embedded Systems Group
+ * 
+ * This code is provided under the terms of the Eclipse Public License (EPL).
+ */
+package de.cau.cs.kieler.klighd.lsp.structuredProgramming.sccharts
+
+import com.google.inject.Inject
+import de.cau.cs.kieler.klighd.internal.util.KlighdInternalProperties
+import de.cau.cs.kieler.klighd.kgraph.KEdge
+import de.cau.cs.kieler.klighd.kgraph.KNode
+import de.cau.cs.kieler.klighd.lsp.KGraphDiagramServer
+import de.cau.cs.kieler.klighd.lsp.KGraphDiagramState
+import de.cau.cs.kieler.klighd.lsp.KGraphLanguageClient
+import de.cau.cs.kieler.klighd.lsp.KGraphLanguageServerExtension
+import de.cau.cs.kieler.klighd.lsp.LSPUtil
+import de.cau.cs.kieler.sccharts.ControlflowRegion
+import de.cau.cs.kieler.sccharts.PreemptionType
+import de.cau.cs.kieler.sccharts.Region
+import de.cau.cs.kieler.sccharts.State
+import de.cau.cs.kieler.sccharts.Transition
+import javax.inject.Singleton
+import org.eclipse.sprotty.Action
+import org.eclipse.xtend.lib.annotations.Accessors
+import org.eclipse.xtext.ide.server.ILanguageServerAccess
+import org.eclipse.xtext.ide.server.ILanguageServerExtension
+import de.cau.cs.kieler.sccharts.impl.SCChartsFactoryImpl
+import java.io.ByteArrayOutputStream
+import org.eclipse.lsp4j.Range
+import org.eclipse.lsp4j.Position
+import java.util.List
+import java.util.Map
+import org.eclipse.lsp4j.TextEdit
+import de.cau.cs.kieler.kexpressions.kext.KExtStandaloneParser
+import de.cau.cs.kieler.kexpressions.keffects.impl.AssignmentImpl
+import de.cau.cs.kieler.kexpressions.impl.ValuedObjectReferenceImpl
+import de.cau.cs.kieler.kexpressions.impl.OperatorExpressionImpl
+import de.cau.cs.kieler.klighd.lsp.ValuedObjectNotFoundException
+import de.cau.cs.kieler.sccharts.extensions.SCChartsTransitionExtensions
+
+/**
+ * Extension responsible for adjusting a model based on the received action message in the actionhandler.
+ * @author felixj
+ */
+@Singleton
+class StructuredProgScchartLanguageServerExtension implements ILanguageServerExtension {
+    // used to change transitions
+    @Inject extension SCChartsTransitionExtensions
+
+    // used to communicate with the frontend eg sending error messages
+    @Accessors KGraphLanguageClient client;
+
+    // used to create new states transitions and regions
+    @Inject
+    SCChartsFactoryImpl factory
+
+    // used to get the client uri
+    @Inject
+    KGraphDiagramState diagramState
+
+    // used to get the model from a given uri
+    @Inject
+    KGraphLanguageServerExtension languageServer
+
+    // used to store the old file size since updating text content replaces old content for now.
+    Position pre_range
+
+    override initialize(ILanguageServerAccess access) {
+        factory = new SCChartsFactoryImpl()
+    }
+
+    /*simply sets the range to be the files size. */
```

Comment:
All the methods below do not have a javadoc but only a comment in front of them.